test/blas1_test.hpp

```diff
@@ -0,0 +1,175 @@
+    }
+  }
+
+  template <typename DataType,
+            typename ValueType = typename DataType::value_type>
+  static void print_cont(const DataType &vec, size_t _N,
+                         std::string name = "vector") {
+    std::cout << name << ": ";
+    for (size_t i = 0; i < _N; ++i) std::cout << vec[i] << " ";
+    std::cout << std::endl;
+  }
+
+  template <typename DataType,
+            typename ValueType = typename DataType::value_type>
+  static buffer<ValueType, 1> make_buffer(DataType &vec) {
+    return buffer<ValueType, 1>(vec.data(), vec.size());
+  }
+
+  template <typename ValueType>
+  static vector_view<ValueType, buffer<ValueType>> make_vview(
+      buffer<ValueType, 1> &buf) {
+    return vector_view<ValueType, buffer<ValueType>>(buf);
+  }
+
+  template <typename DeviceSelector,
+            typename = typename std::enable_if<
+                std::is_same<ExecutorType, SYCL>::value>::type>
+  static cl::sycl::queue make_queue(DeviceSelector s) {
+    return cl::sycl::queue(s, [=](cl::sycl::exception_list eL) {
+      try {
+        for (auto &e : eL) std::rethrow_exception(e);
+      } catch (cl::sycl::exception &e) {
+        std::cout << " E " << e.what() << std::endl;
+      } catch (...) {
+        std::cout << " An exception " << std::endl;
+      }
+    });
+  }
+};
+
+// it is important that all tests are run with the same test size
+// so each time we access this function within the same program, we get the same
+// randomly generated size
+template <class TestClass>
+size_t test_size() {
+  static bool first = true;
+  static size_t N;
+  if (first) {
+    first = false;
+    N = TestClass::rand_size();
+  }
+  return N;
+}
+
+// getting the stride in the same way as the size above
+template <class TestClass>
+size_t test_strd() {
+  static bool first = true;
+  static size_t N;
+  if (first) {
+    first = false;
+    N = ((rand() & 1) * (rand() % 5)) + 1;
+  }
+  return N;
+}
+
+// unpacking the parameters within the test function
+// B is blas_templ_struct
+// TestClass is BLAS1_Test<B>
+// T is default (scalar) type for the test (e.g. float, double)
+// C is the container type for the test (e.g. std::vector)
+// E is the executor kind for the test (sequential, openmp, sycl)
+#define B1_TEST(name) TYPED_TEST(BLAS1_Test, name)
+#define UNPACK_PARAM(test_name)                        \
+  using ScalarT = typename TypeParam::scalar_t;        \
+  using TestClass = BLAS1_Test<TypeParam>;             \
+  using ExecutorType = typename TypeParam::executor_t; \
+  using test = class test_name;
+// TEST_SIZE determines the size based on the suggestion
+#define TEST_SIZE                                                     \
+  ((option_size<test>::value == RANDOM_SIZE) ? test_size<TestClass>() \
+                                             : option_size<test>::value)
+#define TEST_STRD                                                     \
+  ((option_strd<test>::value == RANDOM_SIZE) ? test_strd<TestClass>() \
```

Comment:
If the stride is set to be random, it is intended to be some small random number, otherwise it is a specific value.